src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp

```diff
@@ -77,5 +77,221 @@ void constraint_preserving_bjorhus_corrections_dt_v_zero(
     const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
         four_index_constraint,
     const std::array<DataType, 4>& char_speeds) noexcept;
+
+/// @{
+/*!
+ * \brief Computes the expression needed to set boundary conditions on the time
+ * derivative of the characteristic field \f$v^{0}_{iab}\f$
+ *
+ * \details In the Bjorhus scheme, the time derivatives of evolved variables are
+ * characteristic projected. Constraint-preserving correction terms
+ * \f$T^{\mathrm C}_{ab}\f$, Sommerfeld condition terms for gauge degrees of
+ * freedom \f$T^{\mathrm G}_{ab}\f$, and terms constraining physical degrees of
+ * freedom \f$T^{\mathrm P}_{ab}\f$ are added here to the resulting
+ * characteristic (time-derivative) field:
+ *
+ * \f{align}
+ * \Delta \partial_t v^{-}_{ab} = T^{\mathrm C}_{ab} + T^{\mathrm G}_{ab}
+ *                              + T^{\mathrm P}_{ab}.
+ * \f}
+ *
+ * These terms are given by Eq. (64) of \cite Lindblom2005qh :
+ *
+ * \f{eqnarray}
+ * T^{\mathrm C}_{ab} &=& \frac{1}{2}
+ *      \left(2 k^c k^d l_a l_b - k^c l_b P^d_a - k^c l_a P^d_b - k^d l_b P^c_a
+ *            - k^d l_a P^c_b + P^{cd} P_{ab}\right) \partial_t v^{-}_{cd}\\
+ *     &&+ \frac{1}{\sqrt{2}} \lambda_{-}c^{\hat{0}-}_c \left(l_a l_b k^c +
+ *          P_{ab} l^c - P^c_b l_a - P^c_a l_b \right) \nonumber
+ * \f}
+ *
+ * where \f$l^a (l_a)\f$ is the outgoing null vector (one-form), \f$k^a (k_a)\f$
+ * is the incoming null vector (one-form), \f$P_{ab}, P^{ab}, P^a_b\f$ are
+ * spacetime projection operators defined in `transverse_projection_operator()`,
+ * \f$\partial_t v^{-}_{ab}\f$ is the characteristic projected time derivative
+ * of evolved variables (corresponding to the \f$v^{-}\f$ field), and
+ * \f$c^{\hat{0}\pm}_a\f$ are characteristic modes of the constraint evolution
+ * system:
+ *
+ * \f{align}\nonumber
+ * c^{\hat{0}\pm}_a = F_a \mp n^k C_{ka},
+ * \f}
+ *
+ * where \f$F_a\f$ is the generalized-harmonic (GH) F constraint [Eq. (43) of
+ * \cite Lindblom2005qh], \f$C_{ka}\f$ is the GH 2-index constraint [Eq. (44)
+ * of \cite Lindblom2005qh], and \f$n^k\f$ is the unit spatial normal to the
+ * outer boundary.
+ *
+ * \f{align}
+ * T^{\mathrm G}_{ab} =
```

Comment:
Is there a reference you can give for the gauge terms? I don't think they're given in Lindblom2005. If they don't appear in any particular paper, then it's probably fine to just state the part of SpEC it's based on.

Reply:
yes its indeed Eq. 25 from Rinne et al. Added a citation to it.